src/Integration.Vsix/SLCore/SLCoreLocator.cs

```diff
@@ -31,28 +33,68 @@ internal class SLCoreLocator : ISLCoreLocator
     {
         private const string DefaultPathInsideVsix = "Sloop\\";
         private readonly string basePathInsideVsix;
-        private const string JreSubPath = "jre\\bin\\java.exe";
+        private readonly ISonarLintSettings slSettings;
+        private readonly ILogger logger;
+        private readonly IFileSystem fileSystem;
+        private const string SlCoreJreSubPath = "jre\\bin\\java.exe";
+        private const string WindowsJreSubPath = "bin\\java.exe";
         private const string LibSubPath = "lib\\*";
         private readonly IVsixRootLocator vsixRootLocator;
 
         [ImportingConstructor]
-        public SLCoreLocator(IVsixRootLocator vsixRootLocator) : this(vsixRootLocator, DefaultPathInsideVsix)
+        public SLCoreLocator(IVsixRootLocator vsixRootLocator, ISonarLintSettings sonarLintSettings, ILogger logger) 
+            : this(vsixRootLocator, DefaultPathInsideVsix, sonarLintSettings, logger, new FileSystem())
         {
         }
 
-        internal /* for testing */ SLCoreLocator(IVsixRootLocator vsixRootLocator, string basePathInsideVsix)
+        internal /* for testing */ SLCoreLocator(IVsixRootLocator vsixRootLocator,
+            string basePathInsideVsix,
+            ISonarLintSettings slSettings,
+            ILogger logger,
+            IFileSystem fileSystem)
         {
             this.vsixRootLocator = vsixRootLocator;
             this.basePathInsideVsix = basePathInsideVsix;
+            this.slSettings = slSettings;
+            this.logger = logger;
+            this.fileSystem = fileSystem;
         }
 
         public SLCoreLaunchParameters LocateExecutable()
         {
             var vsixRoot = vsixRootLocator.GetVsixRoot();
 
             //This will be changed later to jre call
-            return new (Path.Combine(vsixRoot, basePathInsideVsix, JreSubPath), 
+            return new (GetJrePath(vsixRoot), 
```

Comment:
I would replace this with `GetCustormJrePath() ?? GetEmbeddedJrePath(vsixRoot)` for more clarity, as it was hard to understand from the first glance that vsixRoot was only used for the embedded case. The logging can be moved from line 76 to line 92